pkg/transformer/kubernetes/k8sutils_test.go

```diff
@@ -0,0 +1,110 @@
+		Args:          []string{"arg1", "arg2"},
+		Volumes:       []string{"/tmp/volume"},
+		Network:       []string{"network1", "network2"}, // not supported
+		Labels:        nil,
+		Annotations:   map[string]string{"abc": "def"},
+		CPUSet:        "cpu_set",            // not supported
+		CPUShares:     1,                    // not supported
+		CPUQuota:      1,                    // not supported
+		CapAdd:        []string{"cap_add"},  // not supported
+		CapDrop:       []string{"cap_drop"}, // not supported
+		Expose:        []string{"expose"},   // not supported
+		Privileged:    true,
+		Restart:       "always",
+		User:          "user", // not supported
+	}
+
+	// An example object generated via k8s runtime.Objects()
+	kompose_object := kobject.KomposeObject{
+		ServiceConfigs: map[string]kobject.ServiceConfig{"app": service},
+	}
+	k := Kubernetes{}
+	objects := k.Transform(kompose_object, kobject.ConvertOptions{CreateD: true, Replicas: 3})
+
+	// Test the creation of the service
+	svc := k.CreateService("foo", service, objects)
+	if svc.Spec.Ports[0].Port != 123 {
+		t.Errorf("Expected port 123 upon conversion, actual %d", svc.Spec.Ports[0].Port)
+	}
+}
+
+/*
+	Test the creation of a service with a specified annotation (kompose.service.type) as "nodeport".
+	The expected result is that Kompose will convert the spec type to "NodePort" upon generation.
+*/
+func TestCreateServiceWithServiceTypeNodePort(t *testing.T) {
+
+	// An example service
+	service := kobject.ServiceConfig{
+		ContainerName: "name",
+		Image:         "image",
+		Environment:   []kobject.EnvVar{kobject.EnvVar{Name: "env", Value: "value"}},
+		Port:          []kobject.Ports{kobject.Ports{HostPort: 123, ContainerPort: 456, Protocol: api.ProtocolTCP}},
+		Command:       []string{"cmd"},
+		WorkingDir:    "dir",
+		Args:          []string{"arg1", "arg2"},
+		Volumes:       []string{"/tmp/volume"},
+		Network:       []string{"network1", "network2"}, // not supported
+		Labels:        nil,
+		Annotations:   map[string]string{"kompose.service.type": "nodeport"},
+		CPUSet:        "cpu_set",            // not supported
+		CPUShares:     1,                    // not supported
+		CPUQuota:      1,                    // not supported
+		CapAdd:        []string{"cap_add"},  // not supported
+		CapDrop:       []string{"cap_drop"}, // not supported
+		Expose:        []string{"expose"},   // not supported
+		Privileged:    true,
+		Restart:       "always",
+		User:          "user", // not supported
+	}
+
+	// An example object generated via k8s runtime.Objects()
+	kompose_object := kobject.KomposeObject{
+		ServiceConfigs: map[string]kobject.ServiceConfig{"app": service},
+	}
+	k := Kubernetes{}
+	objects := k.Transform(kompose_object, kobject.ConvertOptions{CreateD: true, Replicas: 3})
+
+	// Test the creation of the service with modified annotations (kompose.service.type)
+	svc := k.CreateService("foo", service, objects)
+	if svc.Spec.Type != api.ServiceType("NodePort") {
```

Comment:
One last thing ;-)
You test only NodePort, it would be great to test also other options for service.type.
This might be great case for something like this: https://github.com/golang/go/wiki/TableDrivenTests